### PR TITLE
Make workflow curation tasks actually work.

### PR DESCRIPTION
**Pull request by mwoodiupui**
When curation runs in workflow, there was no "current user" and no claimed task, so the code broke when trying to find people to notify about curation failures.

## Description
Set the current user, and find the "flow group" in a different way.

## Instructions for Reviewers
List of changes in this PR:
* Set the current user in the Context used for curation.
* For good measure, disable authorization during curation.  This curation is being run by "the system".
* Reuse existing code to find the current flow step, since it works in this context.

For testing, I enabled the `vscan` task and attached it to a workflow step, then submitted an item containing the EICAR test virus.  DSpace should log a warning about the virus, tell the submitter that "something went wrong" in the UI, and send an email notice that a virus was found in a submission.

## Checklist
- [x] My PR is small in size (e.g. less than 1,000 lines of code, not including comments & integration tests). Exceptions may be made if previously agreed upon.
- [x] My PR passes Checkstyle validation based on the [Code Style Guide](https://wiki.lyrasis.org/display/DSPACE/Code+Style+Guide).
- [x] My PR includes Javadoc for _all new (or modified) public methods and classes_. It also includes Javadoc for large or complex private methods.
- [x] My PR passes all tests and includes new/updated Unit or Integration Tests based on the [Code Testing Guide](https://wiki.lyrasis.org/display/DSPACE/Code+Testing+Guide).
- [x] If my PR includes new libraries/dependencies (in any `pom.xml`), I've made sure their licenses align with the [DSpace BSD License](https://github.com/DSpace/DSpace/blob/main/LICENSE) based on the [Licensing of Contributions](https://wiki.lyrasis.org/display/DSPACE/Code+Contribution+Guidelines#CodeContributionGuidelines-LicensingofContributions) documentation.
- [x] If my PR modifies REST API endpoints, I've opened a separate [REST Contract](https://github.com/DSpace/RestContract/blob/main/README.md) PR related to this change.
- [x] If my PR includes new configurations, I've provided basic technical documentation in the PR itself.
- [x] If my PR fixes an issue ticket, I've [linked them together](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue).

### CODE REVIEW

**Comment by mwoodiupui** [comment]
I should update the manual to emphasize that curation tasks run with authorization disabled when running in workflow.

**Comment by mwoodiupui** [comment]
This patch includes a new method of EPersonService which tries to find "an administrator account" from several sources.  It might be handy elsewhere.

File: dspace-api/src/main/java/org/dspace/eperson/EPersonServiceImpl.java [hunk 1]
```diff
@@ -113,6 +116,21 @@ public EPerson find(Context context, UUID id) throws SQLException {
...
+    public EPerson findAnAdministrator(Context c)
+            throws SQLException {
+        List<EPerson> contacts = groupService.findByName(c, Group.ADMIN).getMembers();
+        EPerson currentUser;
+        if (contacts.isEmpty()) {
+            log.warn("Administrators group is empty");
+            currentUser = findByEmail(c, configurationService.getProperty("mail.admin"));
+            // Null if no such EPerson
+        } else {
+            currentUser = contacts.get(0);
```
Comment (tdonohue) [inline dspace-api/src/main/java/org/dspace/eperson/EPersonServiceImpl.java:129]: @mwoodiupui : Silly question perhaps, but why not just use `mail.admin` (or alternatively `alert.recipient`)?

I'm a little worried about randomly selecting the first administrator in the Admin group, as that gives no real control over where these messages go.  

So, I wonder if this whole method could be removed for simply `configurationService.getProperty("mail.admin");`  (or "alert.recipient")

But, maybe I'm misunderstanding why you aren't just using those email configs?

Comment (tdonohue) [reply dspace-api/src/main/java/org/dspace/eperson/EPersonServiceImpl.java:129]: Oh, I see now... You need an `EPerson` not a simple email address.  That's a bit of a pain.  I'm still finding this approach a little worrisome...there's no guarantee who the first member of the Administrator group is.

**Review by tdonohue — APPROVED** [review]
👍 Thanks @mwoodiupui !  This looks good to me and the code makes sense.  I gave it a basic test by enabling the "requiredmetadata" curation task on the "editstep" of workflow like this in workflow-curation.xml:
```
<taskset-map>
   <mapping collection-handle="default" taskset="checkmetadata" />
</taskset-map>
...
<tasksets>
  <taskset name="checkmetadata">
       <flowstep name="editstep">
         <task name="requiredmetadata">
           <workflow>reject</workflow>
           <notify on="fail">$flowgroup</notify>
           <notify on="fail">$colladmin</notify>
           <notify on="error">$siteadmin</notify>
         </task>
       </flowstep>
  </taskset>
...
</tasksets>
```

From the logs I can see the task was triggered successful and ran.  

I've not been able to test all the functionality that the `vscan` test though, as I've had difficulty getting it working in Docker.  But, I think this basic test of `requiredmetadata` at least proves that curation tasks can be kicked off via workflow steps.

**Comment by tdonohue** [comment]
Because of the refactoring to remove the extra config & simplify it, I think this is `1 Approval` and no longer needs additional docs

**Pull request merged**